### PR TITLE
Issue 2387

### LINKED ISSUE #2387: Outdated dependencies for package.json in root directory
After reading #2386, I checked to see if we had other outdated dependencies and we clearly have many that need to be addressed.

// package.json
```
Package                           Current   Wanted   Latest  Location                                       Depended by
@babel/core                        7.14.3   7.14.3   7.15.8  node_modules/@babel/core                       telescope  
@babel/plugin-transform-runtime   7.13.15  7.13.15   7.15.8  node_modules/@babel/plugin-transform-runtime   telescope  
@babel/preset-env                 7.13.15  7.13.15   7.15.8  node_modules/@babel/preset-env                 telescope  
@babel/preset-react               7.13.13  7.13.13   7.14.5  node_modules/@babel/preset-react               telescope  
@babel/preset-typescript           7.13.0   7.13.0   7.15.0  node_modules/@babel/preset-typescript          telescope  
@elastic/elasticsearch             7.11.0   7.11.0   7.15.0  node_modules/@elastic/elasticsearch            telescope  
@elastic/elasticsearch-mock         0.3.0    0.3.0    0.3.1  node_modules/@elastic/elasticsearch-mock       telescope  
@types/jest                       26.0.20  26.0.20   27.0.2  node_modules/@types/jest                       telescope  
@typescript-eslint/eslint-plugin   4.17.0   4.17.0    5.1.0  node_modules/@typescript-eslint/eslint-plugin  telescope  
@typescript-eslint/parser          4.17.0   4.17.0    5.1.0  node_modules/@typescript-eslint/parser         telescope  
@vercel/node                        1.9.1    1.9.1   1.12.1  node_modules/@vercel/node                      telescope  
@wordpress/wordcount               2.15.0   2.15.0    3.2.2  node_modules/@wordpress/wordcount              telescope  
babel-jest                         26.6.3   26.6.3   27.3.1  node_modules/babel-jest                        telescope  
bull                               3.20.1   3.20.1   3.29.3  node_modules/bull                              telescope  
connect-redis                       5.1.0    5.1.0    6.0.0  node_modules/connect-redis                     telescope  
date-fns                           2.19.0   2.19.0   2.25.0  node_modules/date-fns                          telescope
docker-compose                     0.23.6   0.23.6  0.23.13  node_modules/docker-compose                    telescope
dotenv                              8.2.0    8.2.0   10.0.0  node_modules/dotenv                            telescope
entities                            2.2.0    2.2.0    3.0.1  node_modules/entities                          telescope
eslint                             7.23.0   7.23.0    8.1.0  node_modules/eslint                            telescope
eslint-config-prettier              8.1.0    8.1.0    8.3.0  node_modules/eslint-config-prettier            telescope
eslint-plugin-import               2.22.1   2.22.1   2.25.2  node_modules/eslint-plugin-import              telescope
eslint-plugin-jest-playwright       0.2.1    0.2.1    0.6.0  node_modules/eslint-plugin-jest-playwright     telescope
eslint-plugin-prettier              3.4.0    3.4.0    4.0.0  node_modules/eslint-plugin-prettier            telescope
eslint-plugin-promise               5.1.0    5.1.0    5.1.1  node_modules/eslint-plugin-promise             telescope
eslint-plugin-react                7.22.0   7.22.0   7.26.1  node_modules/eslint-plugin-react               telescope
express-handlebars                  5.3.0    5.3.0    5.3.4  node_modules/express-handlebars                telescope
express-session                    1.17.1   1.17.1   1.17.2  node_modules/express-session                   telescope
express-validator                  6.10.0   6.10.0   6.13.0  node_modules/express-validator                 telescope
fast-xml-parser                    3.19.0   3.19.0   3.20.3  node_modules/fast-xml-parser                   telescope
helmet                              4.4.1    4.4.1    4.6.0  node_modules/helmet                            telescope
highlight.js                       10.6.0   10.6.0   11.3.1  node_modules/highlight.js                      telescope
http-proxy-middleware               1.0.6    1.0.6    2.0.1  node_modules/http-proxy-middleware             telescope
husky                               5.1.3    5.1.3    7.0.4  node_modules/husky                             telescope
ioredis                            4.26.0   4.26.0   4.28.0  node_modules/ioredis                           telescope
ioredis-mock                        5.5.5    5.5.5    5.6.0  node_modules/ioredis-mock                      telescope
jest                               26.6.3   26.6.3   27.3.1  node_modules/jest                              telescope
jest-playwright-preset              1.5.2    1.5.2    1.7.0  node_modules/jest-playwright-preset            telescope
jsdom                              16.5.1   16.5.1   18.0.0  node_modules/jsdom                             telescope
nock                              13.0.11  13.0.11   13.1.4  node_modules/nock                              telescope
node-fetch                          2.6.1    2.6.1    3.0.0  node_modules/node-fetch                        telescope
normalize-url                       6.0.1    6.0.1    7.0.2  node_modules/normalize-url                     telescope
passport                            0.4.1    0.4.1    0.5.0  node_modules/passport                          telescope
passport-saml                       3.1.2    3.1.2    3.2.0  node_modules/passport-saml                     telescope
pino                               6.11.2   6.11.2    7.0.3  node_modules/pino                              telescope
pino-elasticsearch                  5.4.0    5.4.0    6.2.0  node_modules/pino-elasticsearch                telescope
pino-pretty                         4.7.1    4.7.1    7.1.0  node_modules/pino-pretty                       telescope
playwright                         1.11.0   1.11.0   1.16.0  node_modules/playwright                        telescope
prettier                            2.2.1    2.2.1    2.4.1  node_modules/prettier                          telescope
pretty-quick                        3.1.0    3.1.0    3.1.1  node_modules/pretty-quick                      telescope
sanitize-html                       2.3.2    2.3.2    2.5.2  node_modules/sanitize-html                     telescope
supertest                           6.1.3    6.1.3    6.1.6  node_modules/supertest                         telescope
ts-jest                            26.5.4   26.5.4   27.0.7  node_modules/ts-jest                           telescope
```

We have other package.json files but I figured it would be too much to do for one issue so let's break it down into multiple issues.

### PR DESCRIPTION
**Pull request by LuigiZaccagnini**
## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

- Updated the root package.json packages. 
- Added a global textEncoder for upgrade jsdom

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: package.json
Comment (humphd) [inline package.json:58]: This should be removed, we don't use it anymore.

File: package.json
Comment (humphd) [inline package.json:59]: Filed #2504 to remove this.  We don't need it.

**Review by humphd — CHANGES_REQUESTED** [review]
Looks good, tests are all passing.  One small fix.

File: package.json
Comment (humphd) [reply package.json]: You can't remove this withing fixing the other things I mention in #2504.  Please add `body-parser` back, and read review comments more carefully.

File: package.json
Comment (humphd) [inline package.json:58]: I told you the wrong thing, I forgot the status service uses this.  We better leave this in.

File: src/api/status/public/index.html
Comment (humphd) [inline src/api/status/public/index.html:42]: This file needs to be rebased to pick up the changes in master.

File: src/backend/data/stats.js
Comment (humphd) [inline src/backend/data/stats.js:1]: Revert this, sorry about that.

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by Qiwen-Yu** [comment]
Mark this one as the code review template and example to me.

File: src/backend/data/stats.js
Comment (humphd) [reply src/backend/data/stats.js:1]: This hasn't been reverted.  Please do

```
git checkout master src/backend/data/stats.js
```

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/backend/data/stats.js
Comment (LuigiZaccagnini) [reply src/backend/data/stats.js]: Just fixed this in the recent commit!

**Review by humphd — DISMISSED** [review]

**Review by manekenpix — COMMENTED** [review]
Please don't merge master into your branch. Revert that change and rebase.

**Review by manekenpix — CHANGES_REQUESTED** [review]
I'll block this until the PR gets rebased.

**Review by humphd — DISMISSED** [review]

**Comment by humphd** [comment]
Can we get another review on this?

**Review by manekenpix — APPROVED** [review]

**Comment by humphd** [comment]
@LuigiZaccagnini this needs another rebase:

```sh
git checkout master
git pull upstream master
git checkout issue-2387
git rebase master
# fix merge conflict in package.json
git push origin issue-2387 -f
```

Then ping me so I can review and merge.

**Comment by humphd** [comment]
What's happening with this?

**Comment by humphd** [comment]
This is out of date with `package.json`, please fix.

**Review by humphd — APPROVED** [review]
I don't understand the changes to the non-package.json files in this.  Why would prettier change things?

I *think* the rest is good.  I'll add @menghif as a reviewer, since this will cause churn in #2545

**Comment by LuigiZaccagnini** [comment]
The change in `src/backend/utils/html/dom.js ` was done because the updates in the package.json required an encoder for JSDom. The other changes were prettier as they weren't passing the tests on my end with the rebase.

**Comment by humphd** [comment]
OK, sounds good.

**Review by menghif — APPROVED** [review]
Looks good! This will impact #2563 but seems like no one is working on it yet.

**Comment by humphd** [comment]
OK, going to merge this.  FYI @HyperTHD, @manekenpix, in case this breaks staging in ways we haven't tested.

**Pull request merged**
